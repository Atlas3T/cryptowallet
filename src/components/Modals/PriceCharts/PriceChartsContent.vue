<template>
  <div>
    <div
      :class="{ active: loading }"
      class="sending-spinner-overlay"
    >
      <Spinner />
    </div>
    <div
      v-if="wallet"
      class="header-section"
    >
      <div class="header-back-button-wrapper">
        <q-btn
          icon="arrow_back"
          size="lg"
          class="icon-btn back-arrow-btn"
          flat
          @click.prevent="goBack"
        />
      </div>

      <div>
        <img
          :src="coinLogo"
          class="coin-logo"
        >
        {{ wallet.displayName }}
      </div>

      <div
        class="header-settings-button-wrapper"
      >
        <q-btn
          icon="refresh"
          color="secondary"
          size="lg"
          class="icon-btn icon-btn-right"
          flat
          @click="loadData"
        />
      </div>
    </div>
    <div
      v-if="wallet && latestPrice"
      class="modal-layout-wrapper"
    >
      <div class="price-info justify-center">
        <div class="row labels">
          <div class="col-6">
            1 {{ coinSymbol }} / {{ selectedCurrency.code }}
          </div>
          <div
            v-if="latestPrice.data.VOLUME24HOURTO !== 0"
            class="col-6"
          >
            {{ $t('volume24h') }} {{ selectedCurrency.code }}
          </div>
        </div>
        <div class="row price">
          <div class="col-6">
            {{ selectedCurrency.symbol }}{{ latestPrice.data.PRICE.toFixed(2) }}
          </div>
          <div
            v-if="latestPrice.data.VOLUME24HOURTO !== 0"
            class="col-6"
          >
            {{ selectedCurrency.symbol }}{{ latestPrice.data.VOLUME24HOURTO.toFixed(0) }}
          </div>
        </div>
        <div
          :style="{ color: percentColor}"
          class="row"
        >
          <div class="col-6">
            {{ latestPrice.data.CHANGEPCT24HOUR.toFixed(2) }}%
          </div>
        </div>
      </div>
      <PriceChart
        v-if="(chartDataExists || showChart)"
        :gradient="gradientStroke"
      />
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import PriceChart from '@/components/PriceCharts/ChartContainer';
import Spinner from '@/components/Spinner';
import Coin from '@/store/wallet/entities/coin';
import IconList from '@/assets/cc-icons/icons-list.json';

export default {
  name: 'PriceCharts',
  components: {
    PriceChart,
    Spinner,
  },
  data() {
    return {
      gradientStroke: '',
      loading: false,
      showChart: false,
    };
  },
  computed: {
    ...mapState({
      id: (state) => { return state.route.params.id; },
    }),
    wallet() {
      return this.$store.getters['entities/wallet/find'](this.id);
    },
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    supportedCoins() {
      return Coin.all();
    },
    coinSymbol() {
      return this.supportedCoins.find((coin) => { return coin.name === this.wallet.name; }).symbol;
    },
    latestPrice() {
      const prices = this.$store.getters['entities/latestPrice/find'](`${this.coinSymbol}_${this.selectedCurrency.code}`);
      if (!prices) {
        return null;
      }
      return prices;
    },
    percentColor() {
      const priceChange = this.latestPrice.data.CHANGEPCT24HOUR;
      if (priceChange > 0) {
        return '#00FF00';
      }
      return '#de4662';
    },
    coinLogo() {
      if (IconList.find((icon) => { return icon.symbol === this.wallet.symbol.toUpperCase(); })) {
        return `./statics/cc-icons/color/${this.wallet.symbol.toLowerCase()}.svg`;
      }
      return './statics/cc-icons/color/generic.svg';
    },
    chartDataExists() {
      const day = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_day`);
      const week = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_week`);
      const month = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_month`);
      if (!day || !week || !month) {
        return false;
      }

      return true;
    },

  },
  async mounted() {
    this.loadData();
  },
  methods: {
    async loadData() {
      this.showChart = false;
      this.loading = true;

      try {
        await this.backEndService.loadCoinPriceData(this.coinSymbol);
      } catch (err) {
        this.errorHandler(err);
      }

      this.loading = false;
      this.showChart = true;
    },
    goBack() {
      this.$router.go(-1);
    },
  },
};
</script>

<style lang="scss">
.close-btn .q-btn__content .modal-layout-wrapper{
  justify-content: flex-start;
}

.modal-layout-wrapper{
  padding: 1em;
  margin-top: "10px";
}

.q-toolbar{
  background-color: #1e3c57;
}

.price-info{
  margin-top: 15px;
  margin-bottom: 10px;
  padding: 0 0.5em;
}

.price-info .row{
  padding-bottom: 5px;
}

.row .price-percent{
  color: #de4662;
}

.row .price{
  font-size: x-large;
  color: white;
}

.row .labels{
  color: sandybrown;
}

.sending-spinner-overlay {
 position: absolute;
 height: 100vh;
 width: 100vw;
 z-index: 0;
 display: flex;
 align-items: center;
 justify-content: center;
 background: rgba(30, 60, 87, 0.9);
 opacity: 0;
 transition: all ease-in-out 250ms;
}

.sending-spinner-overlay.active{
 opacity: 1;
 z-index: 2;
}
</style>
