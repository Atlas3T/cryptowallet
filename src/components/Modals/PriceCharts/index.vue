<template>
  <div>
    <q-modal
      v-model="priceChartModalOpened"
      class="dark-modal modal"
    >
      <div
        :class="{ active: loading }"
        class="sending-spinner-overlay"
      >
        <Spinner />
      </div>
      <div
        v-if="wallet"
        class="header-section"
      >
        <div class="header-back-button-wrapper">
          <q-btn
            icon="arrow_back"
            size="lg"
            class="icon-btn back-arrow-btn"
            flat
            @click.prevent="goBack"
          />
        </div>

        <div>
          <img
            :src="coinLogo"
            class="coin-logo"
          >
          {{ wallet.displayName }}
        </div>

        <div
          class="header-settings-button-wrapper"
        >
          <q-btn
            icon="refresh"
            color="secondary"
            size="lg"
            class="icon-btn icon-btn-right"
            flat
            @click="loadData"
          />
        </div>
      </div>
      <div
        v-if="wallet && latestPrice"
        class="modal-layout-wrapper"
      >
        <div class="price-info justify-center">
          <div class="row labels">
            <div class="col-6">
              1 {{ coinSymbol }} / {{ selectedCurrency.code }}
            </div>
            <div
              v-if="latestPrice.data.VOLUME24HOURTO !== '£ 0'"
              class="col-6"
            >
              {{ $t('volume24h') }}{{ selectedCurrency.code }}
            </div>
          </div>
          <div class="row price">
            <div class="col-6">
              {{ latestPrice.data.PRICE }}
            </div>
            <div
              v-if="latestPrice.data.VOLUME24HOURTO !== '£ 0'"
              class="col-6"
            >
              {{ latestPrice.data.VOLUME24HOURTO }}
            </div>
          </div>
          <div
            :style="{ color: percentColor}"
            class="row"
          >
            <div class="col-6">
              {{ latestPrice.data.CHANGEPCT24HOUR }}%
            </div>
          </div>
        </div>
        <PriceChart
          v-if="priceChartModalOpened && (showChart || chartDataExists)"
          :gradient="gradientStroke"
        />
      </div>
    </q-modal>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import PriceChart from '@/components/PriceCharts/ChartContainer';
import Spinner from '@/components/Spinner';
import Prices from '@/store/prices';
import Latest from '@/store/latestPrice';
import Coin from '@/store/wallet/entities/coin';
import IconList from '@/assets/cc-icons/icons-list.json';


export default {
  name: 'PriceCharts',
  components: {
    PriceChart,
    Spinner,
  },
  data() {
    return {
      priceChartModalOpened: false,
      gradientStroke: '',
      loading: false,
      showChart: false,
    };
  },
  computed: {
    ...mapState({
      id: (state) => { return state.route.params.id; },
      authenticatedAccount: (state) => { return state.settings.authenticatedAccount; },
    }),
    wallet() {
      return this.$store.getters['entities/wallet/find'](this.id);
    },
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    supportedCoins() {
      return Coin.all();
    },
    coinSymbol() {
      return this.supportedCoins.find((coin) => { return coin.name === this.wallet.name; }).symbol;
    },
    latestPrice() {
      const prices = this.$store.getters['entities/latestPrice/find'](`${this.coinSymbol}_${this.selectedCurrency.code}`);
      if (!prices) {
        return null;
      }
      return this.$store.getters['entities/latestPrice/find'](`${this.coinSymbol}_${this.selectedCurrency.code}`);
    },
    percentColor() {
      const priceChange = this.latestPrice.data.CHANGEPCT24HOUR;
      if (priceChange.charAt(0) !== '-') {
        return 'lime';
      }
      return '#de4662';
    },
    coinLogo() {
      if (IconList.find((icon) => { return icon.symbol === this.wallet.symbol.toUpperCase(); })) {
        return `./statics/cc-icons/color/${this.wallet.symbol.toLowerCase()}.svg`;
      }
      return './statics/cc-icons/color/generic.svg';
    },
    chartDataExists() {
      const day = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_day`);
      const week = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_week`);
      const month = this.$store.getters['entities/prices/find'](`${this.coinSymbol}_${this.selectedCurrency.code}_month`);
      if (!day || !week || !month) {
        return false;
      }
      return true;
    },
  },
  watch: {
    $route: {
      handler(to) {
        if (to.name === 'coinPrices' || to.name === 'coinSinglePrices') {
          this.priceChartModalOpened = true;
        } else {
          this.priceChartModalOpened = false;
        }
      },
    },
    priceChartModalOpened: {
      handler(newVal, oldVal) {
        if (oldVal === true && newVal === false) {
          if (this.$store.state.route.name === 'coinPrices'
              || this.$store.state.route.name === 'coinSinglePrices') {
            this.$router.go(-1);
          }
        }
      },
    },
  },
  async mounted() {
    this.$root.$on('priceChartModalOpened', (value) => {
      if (value === true) {
        this.priceChartModalOpened = value;
        this.loadData();
      }
    });
  },
  methods: {
    viewChart() {
      this.priceChartModalOpened = true;
    },
    async loadData() {
      this.showChart = false;
      this.loading = true;

      let coinSDK = this.coinSDKS[this.wallet.sdk];
      if (this.wallet.sdk === 'ERC20') {
        coinSDK = this.coinSDKS[this.wallet.parentSdk];
      }

      try {
        const dayData = await coinSDK.getHistoricalData(this.coinSymbol, this.selectedCurrency.code, 'day');
        const weekData = await coinSDK.getHistoricalData(this.coinSymbol, this.selectedCurrency.code, 'week');
        const monthData = await coinSDK.getHistoricalData(this.coinSymbol, this.selectedCurrency.code, 'month');
        const latestPrice = await coinSDK.getPriceFeed(
          [this.coinSymbol],
          [this.selectedCurrency.code],
        );

        const checkExists = (period, data) => {
          const price = Prices.find([`${this.coinSymbol}_${this.selectedCurrency.code}_${period}`]);
          if (!price) {
            Prices.$insert({
              data: {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
                period,
                updated: +new Date(),
                data,
              },
            });
            return false;
          }
          return true;
        };

        const wherePrice = (record, item) => {
          return (
            record.coin === item.coin
            && record.currency === item.currency
            && record.period === item.period
          );
        };

        if (checkExists('day', dayData)) {
          Prices.$update({
            where: (record) => {
              return wherePrice(record, {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
                period: 'day',
              });
            },
            data: {
              updated: +new Date(),
              data: dayData,
            },
          });
        }
        if (checkExists('week', weekData)) {
          Prices.$update({
            where: (record) => {
              return wherePrice(record, {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
                period: 'week',
              });
            },
            data: {
              updated: +new Date(),
              data: weekData,
            },
          });
        }
        if (checkExists('month', monthData)) {
          Prices.$update({
            where: (record) => {
              return wherePrice(record, {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
                period: 'month',
              });
            },
            data: {
              updated: +new Date(),
              data: monthData,
            },
          });
        }
        const checkPriceExists = (symbol, data) => {
          const price = Latest.find([`${symbol}_${this.selectedCurrency.code}`]);
          if (!price) {
            Latest.$insert({
              data: {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
                updated: +new Date(),
                data,
              },
            });
            return false;
          }
          return true;
        };

        const whereLatest = (record, item) => {
          return (
            record.coin === item.coin
            && record.currency === item.currency
          );
        };

        const latest = latestPrice[this.coinSymbol][this.selectedCurrency.code];
        if (checkPriceExists(this.coinSymbol, latest)) {
          Latest.$update({
            where: (record) => {
              return whereLatest(record, {
                coin: this.coinSymbol,
                currency: this.selectedCurrency.code,
              });
            },
            data: {
              updated: +new Date(),
              data: latestPrice[this.coinSymbol][this.selectedCurrency.code],
            },
          });
        }
      } catch (err) {
        this.errorHandler(err);
      }

      this.loading = false;
      this.showChart = true;
      return false;
    },
    goBack() {
      this.$router.go(-1);
    },
  },
};
</script>

<style lang="scss">
.close-btn .q-btn-inner .modal-layout-wrapper{
  justify-content: flex-start;
}

.modal-layout-wrapper{
  padding: 1em;
  margin-top: "10px";
}

.q-toolbar{
  background-color: #1e3c57;
}

.price-info{
  margin-top: 15px;
  margin-bottom: 10px;
  padding: 0 0.5em;
}

.price-info .row{
  padding-bottom: 5px;
}

.row .price-percent{
  color: #de4662;
}

.row .price{
  font-size: x-large;
  color: white;
}

.row .labels{
  color: sandybrown;
}

.sending-spinner-overlay {
 position: absolute;
 height: 100vh;
 width: 100vw;
 z-index: 0;
 display: flex;
 align-items: center;
 justify-content: center;
 background: rgba(30, 60, 87, 0.9);
 opacity: 0;
 transition: all ease-in-out 250ms;
}

.sending-spinner-overlay.active{
 opacity: 1;
 z-index: 2;
}
</style>
