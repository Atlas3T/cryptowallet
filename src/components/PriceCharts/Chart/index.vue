<script>
// cannot fix this, _chart is coming from vue-chartjs
/* eslint-disable vue/no-reserved-keys */
/* eslint-disable no-underscore-dangle */
import { Line, mixins } from 'vue-chartjs';

const { reactiveProp } = mixins;

export default {
  extends: Line,
  mixins: [reactiveProp],
  props: {
    options: {
      type: Object,
      default: null,
    },
    chartData: {
      type: Object,
    },
  },
  data() {
    return {
      _chart: '',
    };
  },
  computed: {
    gradient() {
      const colour = 450;
      const newGradient = this.$refs.canvas.getContext('2d').createLinearGradient(0, 0, 0, colour);
      const transparency = 0.3;
      newGradient.addColorStop(0, 'rgba(250, 188, 87, 0.5)');
      newGradient.addColorStop(transparency, 'rgba(250, 188, 87, 0.25)');
      newGradient.addColorStop(1, 'rgba(250, 188, 87, 0)');

      return newGradient;
    },
  },
  watch: {
    chartData: {
      handler() {
        this.loadChart();
      },
    },
  },
  mounted() {
    this.loadChart();
  },
  methods: {
    newLegend() {
      this.$emit('legendHtml', this.chart);
    },
    loadChart() {
      this.chartData.datasets.forEach((set) => {
        set.backgroundColor = this.gradient;
      });
      this.renderChart(this.chartData, this.options);
      this.chart = this.$data._chart;
      this.newLegend();
    },
  },
};
</script>
<style>
</style>
