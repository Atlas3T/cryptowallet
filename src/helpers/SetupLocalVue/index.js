import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
