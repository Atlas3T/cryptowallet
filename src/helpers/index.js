import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
