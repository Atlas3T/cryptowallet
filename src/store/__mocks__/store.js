import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
