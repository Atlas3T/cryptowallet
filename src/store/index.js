/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
