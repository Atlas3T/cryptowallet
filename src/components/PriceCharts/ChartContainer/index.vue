<template>
  <div class="chart-container">
    <div class="row justify-center">
      <q-btn
        class="chart-button"
        color="secondary"
        label="24H"
        @click="onClick(0)"
      />
      <q-btn
        class="chart-button"
        color="secondary"
        label="1W"
        @click="onClick(1)"
      />
      <q-btn
        class="chart-button"
        color="secondary"
        label="1M"
        @click="onClick(2)"
      />
    </div>

    <Chart
      v-if="chartData"
      :chart-data="chartData"
      :options="options"
      @legendHtml="updateLegend"
    />
  </div>
</template>

<script>
import Chart from '@/components/PriceCharts/Chart';

export default {
  name: 'ChartContainer',
  components: { Chart },
  props: {
    datasets: {
      type: Array,
      required: true,
    },
  },
  data() {
    return {
      newChart: '',
    };
  },
  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    chartData() {
      return {
        datasets: [{
          label: '24H',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          data: this.datasets[0].data,
        },
        {
          label: '1W',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          hidden: true,
          data: this.datasets[1].data,
        },
        {
          label: '1M',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          hidden: true,
          data: this.datasets[2].data,
        }],
      };
    },
    options() {
      return {
        animations: {
          duration: 2000,
        },
        layout: {
          padding: {
            top: 10,
          },
        },
        scales: {
          xAxes: [{
            type: 'time',
            ticks: {
              minRotation: '45',
              fontColor: '#78d2e6',
              padding: 15,

            },
            gridLines: {
              display: false,
            },
          }],
          yAxes: [{
            position: 'left',
            color: '#78d2e6',
            ticks: {
              fontColor: '#78d2e6',
              callback: (value) => { return `${this.selectedCurrency.symbol}${value}`; },
            },
          }],
        },
        legend: {
          display: false,
        },

      };
    },
  },
  methods: {
    updateLegend(chart) {
      this.newChart = chart;
    },

    onClick(index) {
      const chart = this.newChart;
      [0, 1, 2].forEach((i) => {
        if (i !== index) {
          chart.getDatasetMeta(i).hidden = true;
        }
      });
      chart.getDatasetMeta(index).hidden = false;
      chart.update();
      this.$emit('update', index);
    },
  },
};
</script>

<style>
.chart-button{
    margin-right: 10px;
    margin-left: 10px;
    margin-bottom: 15px;
    margin-top: 10px;
    min-height: 1.75em;
}
.chart-container{
    margin: 0;
}

</style>
