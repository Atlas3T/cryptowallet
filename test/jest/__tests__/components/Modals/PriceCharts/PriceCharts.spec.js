import PriceCharts from '@/components/Modals/PriceCharts/PriceChartsContent.vue';
import { shallowMount } from '@vue/test-utils';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
// import Prices from '@/store/prices';
// import Latest from '@/store/latestPrice';
// import Coin from '@/store/wallet/entities/coin';

describe('modals/PriceCharts', () => {
  let storeMocks;
  let wrapper;
  let router;

  function wrapperInit(options) {
    return shallowMount(PriceCharts, options);
  }

  function storeInit(custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: '/wallet/' });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      store: storeMocks.store,
    });
  }
  beforeEach(() => {
    jest.clearAllMocks();
    return storeInit();
  });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
