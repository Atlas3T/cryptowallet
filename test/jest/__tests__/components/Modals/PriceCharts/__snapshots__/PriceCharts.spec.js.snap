// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modals/PriceCharts renders and matches snapshot 1`] = `
<div>
  <div
    class="sending-spinner-overlay"
  >
    <div
      class="lds-ring"
    >
      <div />
       
      <div />
       
      <div />
       
      <div />
    </div>
  </div>
   
  <div
    class="header-section"
  >
    <div
      class="header-back-button-wrapper"
    >
      <button
        class="q-btn inline q-btn-item non-selectable icon-btn back-arrow-btn q-btn--flat q-btn--rectangle q-focusable q-hoverable"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        <div
          class="q-focus-helper"
          tabindex="-1"
        />
        <div
          class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
        >
          <i
            aria-hidden="true"
            class="material-icons q-icon"
          >
            arrow_back
          </i>
        </div>
      </button>
    </div>
     
    <div>
      <h1
        class="header-h1"
      >
        <img
          class="coin-logo"
          src="./statics/cc-icons/color/btc.svg"
        />
        
        Bitcoin
      
      </h1>
    </div>
     
    <div
      class="header-settings-button-wrapper"
    >
      <button
        class="q-btn inline q-btn-item non-selectable icon-btn icon-btn-right q-btn--flat q-btn--rectangle text-secondary q-focusable q-hoverable"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        <div
          class="q-focus-helper"
          tabindex="-1"
        />
        <div
          class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
        >
          <i
            aria-hidden="true"
            class="material-icons q-icon"
          >
            refresh
          </i>
        </div>
      </button>
    </div>
  </div>
   
  <div
    class="modal-layout-wrapper"
  >
    <div
      class="price-info justify-center"
    >
      <div
        class="row labels"
      >
        <div
          class="col-6"
        >
          
          1 BTC / GBP
        
        </div>
      </div>
       
      <div
        class="row price"
      >
        <div
          class="col-6"
        >
          
          Â£3818.12
        
        </div>
      </div>
       
      <div
        class="row"
        style="color: rgb(0, 255, 0);"
      >
        <div
          class="col-6"
        >
          
          4.73%
        
        </div>
      </div>
    </div>
     
    <pricechart-stub />
     
    <div
      class="row justify-center"
    >
      
      Last Updated: 2019-4-3 16:37:49
    
    </div>
  </div>
</div>
`;
